src/dmd/parse.d

```diff
@@ -7664,6 +7664,15 @@ final class Parser(AST) : Lexer
                 if (token.value == TOK.leftParentheses)
                 {
                     nextToken();
+                    if (token.value == TOK.identifier && peekNext() == TOK.leftParentheses)
+                    {
+                        error(loc, "unexpected `(` after `%s`, inside `is` expression. Try nesting the contents of `is` with a `typeof` expression", token.toChars());
+                        nextToken();
+                        Token* tempTok = peekPastParen(&token);
+                        memcpy(&token, tempTok, Token.sizeof);
```

Comment:
What is the purpose of the memcpy()?

Reply:
ldc forbids assigning to a token. before the mempcy the code looked like this:

```d
nextToken();
token = *peekPastParens(&token);
```

I did this in order to avoid further error messages. I don't know how to do it otherwise that to just call nextToken() until the current token matches the one that peekPastParens points to.

Reply:
I meant what is the purpose of assigning to token? Lookahead is used elsewhere in the parser to decide what is being parsed.

Reply:
The purpose of assigning to a token is to skip the list of parameters that is passed to the function call inside the is expression

Reply:
> ldc forbids assigning to a token. before the mempcy the code looked like this:

Just tested this locally and LDC 1.10 compiles dmd just fine. Maybe you weren't using the latest LDC?